### DISCUSSION TITLE
Validate on blur

### DISCUSSION THREAD

**Discussion by doogdeb** [sinanbozkus/FormHelper#19] [Q&A] [answered]
Hi, I can see that it validates client-side on submit; but can it be configured to validate fields on blur as well?

**Comment by sinanbozkus** [comment]
> Hi, I can see that it validates client-side on submit; but can it be configured to validate fields on blur as well?

What kind of validate do you want to add? Can you give me an example? (Like required field, email alredy in use check etc.)

**Reply by doogdeb** [reply]
For any type of fluent validation (required, less than, equal to, custom etc.). Unobtrusive validation fires on blur, so If user tabs through the fields the validation will fire. Can Form Helper be configured to do the same?

**Reply by sinanbozkus** [reply]
I'm thinking about this. I am going to find a solution soon.

**Reply by doogdeb** [reply]
> I'm thinking about this. I am going to find a solution soon.

That's great. That's the only thing that would stop me from being able to implement this in work.

**Reply by sinanbozkus** [reply]
> For any type of fluent validation (required, less than, equal to, custom etc.). Unobtrusive validation fires on blur, so If user tabs through the fields the validation will fire. Can Form Helper be configured to do the same?

https://im5.ezgif.com/tmp/ezgif-5-26ba596a1eba.gif

FormHelper already supports validations like "required", "less than" and "equal to" on blur/keyup. You can check the Gif or the sample project. 

If you want to do custom validates like "email address is already in use". I can add an attribute to inputs like "asp-custom-validate-url". Every input can go to an action to validate own value. Is it ok for you?

**Reply by doogdeb** [reply]
I've created a sample project based off your blog post. When page first loads and fields are empty, tabbing out doesn't fire validation. Only on populating field and deleting does it then fire. I just have NotNull validation on the field.

**Comment by sinanbozkus** [comment, marked as answer]
I found a solution and published a new prerelease package.

https://www.nuget.org/packages/FormHelper/3.3.1-beta2

After tests, I am going to add this feature to the new version. Can you check it?

Thanks.

**Reply by doogdeb** [reply]
Hi, I pulled down your latest package but unfortunately it's same issue as before.

**Reply by doogdeb** [reply]
Just to confirm I'm running FluentValidation..AspNetCore 9..5.0 and FormHelper 3.3.1 beta 2.

**Reply by sinanbozkus** [reply]
https://im5.ezgif.com/tmp/ezgif-5-24e24bc4ff9c.gif

Look, it's working. Are you using the embedded javascript bundle file? ("/formhelper/formhelper.bundle.min.js" from nuget package)

**Reply by doogdeb** [reply]
These are the scripts.

<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="/formhelper/formhelper.bundle.min.js"></script>

**Reply by doogdeb** [reply]
Hi, it was related to the browser caching the js file. I opened up in Private mode and can confirm it is now working as expected.

**Reply by doogdeb** [reply]
Thanks for fixing so quickly.

**Reply by doogdeb** [reply]
I have another query but I'll create another discussion. for it.

**Reply by sinanbozkus** [reply]
> Thanks for fixing so quickly.

You're welcome. I would be happy if you give a star to this library. 
And you can mark an answer to the message.
Thanks.

**Reply by doogdeb** [reply]
Of course. I just gave a star.

**Reply by doogdeb** [reply]
One thing to note, is that because the script wasn't versioned, the browser cached it. Will cause issues for users who update package without changing script version.

<script src="/formhelper/formhelper.bundle.min.js"></script>
<script src="/formhelper/formhelper.bundle.min.js?v2"></script>
